✨(metabase) add `maps` data and notebook (#144)
MTES-MCT/qualicharge: src/notebook/misc/maps.md

---
jupyter:
  jupytext:
    text_representation:
      extension: .md
      format_name: markdown
      format_version: '1.3'
      jupytext_version: 1.16.2
  kernelspec:
    display_name: Python 3 (ipykernel)
    language: python
    name: python3
---

# QualiCharge maps

## Introduction

> L'utilisation de cartes choroplèthe pour la restitution d'indicateurs QualiCharge nécessite le chargement préalable des contours des zones affichées regroupées par cartes.

Reviewer: ```suggestion
L'utilisation de cartes choroplèthes pour la restitution d'indicateurs QualiCharge nécessite le chargement préalable des contours des zones affichées regroupées par cartes.
```

> Pour des raisons de facilité de consultation, les cartes utilisées sont séparées par zone géographique:

Reviewer: ```suggestion
Pour des raisons de facilité de consultation, les cartes utilisées sont séparées par zone géographique :
```

- Territoire européen (régions européennes : 11, 24, 27, 28, 32, 44, 52, 53, 75, 76, 84, 93, 94)
- Territoire américain (régions Guadeloupe, Martinique, Guyane : 1, 2, 3)
- Territoire africain (régions La Réunion, Mayotte : 4, 6)

> Quatre niveaux de découpage sont proposés:

Reviewer: ```suggestion
Quatre niveaux de découpage sont proposés :
```

- régions
- départements
- communes
- agglomérations

Le quatrième niveau est dérivé du découpage communal en intégrant les regroupements de communes définis dans les établissement public de coopération intercommunale (EPCI) à ficalité propre (métropoles, communautés urbaines, communautés d'agglomération, communautés de communes, Aix-Marseille et Grand Paris).

Les régions sont découpées en départements. Les régions ultramarines (hors territoire européen) ne comportent qu'un seul département (code '97x' avec 'x' code région).
Les départements sont découpés en communes.

> Les agglomérations sont des regroupements de communes

Reviewer: ```suggestion
Les agglomérations sont des regroupements de communes.
```

## Map utilisées

> Les cartes utilisées pour restituer les indicateurs sont alors les suivantes (la métropole correspond au territoire européen):

Reviewer: ```suggestion
Les cartes utilisées pour restituer les indicateurs sont alors les suivantes (la métropole correspond au territoire européen) :
```

- Territoire:
    - métropole par régions
    - métropole par départements
- 18 régions métropolitaines et ultramarines (pour chaque région):
    - région par départements
    - région par ECPI
    - région par commune

A ces cartes sont ajoutées les cartes des collectivités d'outre-mer (codes : 975, 977, 978, 986, 987) et des territoires à statut particulier (codes : 984, 988, 989). Ces structures ne disposent pas d'EPCI.

## Fichiers associés

Les fichiers GeoJson pour chaque map sont extraits des [contours administratifs Etalab](https://etalab-datasets.geo.data.gouv.fr/contours-administratifs/2024/geojson/). 

  - metropole_reg.geojson
  - metropole_dep.geojson
  - region_xx_dep.geojson
  - region_xx_epci.geojson
  - region_xx_com.geojson

  avec xx : code région

> Chaque contour (type : Polygon ou MultiPolygon) est associé à deux propriétés:

Reviewer: ```suggestion
Chaque contour (type : Polygon ou MultiPolygon) est associé à deux propriétés:

```

                                    for row in list(df.itertuples()) if row[reg] in reg_metro]}
    print(ext)
    file_metro = 'metropole' + ext +'.geojson'
    with open(maps_dir + file_metro, 'w', encoding ='utf8') as map_file:
        json.dump(map_geo, map_file)
    
    print(map_geo['features'][0]['properties'])
```

### Régions

```python
import pandas as pd

queries_reg = [query_dep, query_ecpi, query_city]
extension = ['_dep', '_epci', '_com']

with engine.connect() as conn:
    df = pd.read_sql_query(query_dep, conn)
for query, ext in zip(queries_reg, extension):
    with engine.connect() as conn:
        df = pd.read_sql_query(query, conn)
    for reg in regions:
        file_dep = 'region_' + reg + ext +'.geojson'
        map_geo = {'type': 'FeatureCollection', 
                   'features': [{'type': 'Feature', 
                                 'properties': {'code': row[1], 'nom': row[2]}, 
                                 'geometry': {'type': ('MultiPolygon' if str(row[3])[:4] == '[[[[' else 'Polygon'), 'coordinates': row[3]}} 

>                                 for row in list(df.itertuples()) if row[4] == reg]}

Reviewer: Maybe the following would more readable:

```suggestion
                                for row in df[df["code"] == reg].itertuples()}
```
